server/src/__tests__/supplier.test.ts

```diff
@@ -0,0 +1,188 @@
+import request from "supertest";
+import connection from "../db/connection";
+import app from "../app";
+
+const HOME_ROUTE = "/suppliers";
+
+describe("Operations on the supplier route", () => {
+  beforeAll(async () => {
+    await connection.create();
+  });
+
+  afterAll(async () => {
+    await connection.close();
+  });
+
+  describe("Creating a new supplier", () => {
+    describe("Should return error on invalid data", () => {
+      it("Should return 400 if no name is sent to the route", async () => {
+        const res = await request(app).post(HOME_ROUTE).send({});
+
+        expect(res.statusCode).toBe(400);
+        expect(res.body).toHaveProperty("errorKey");
+        expect(res.body).toHaveProperty("errorDescription");
+        expect(res.body.errorKey).toBe("name");
+        expect(res.body.errorDescription).toBe("name is required");
+      });
+
+      it("Should return 400 if no phone is sent to the route", async () => {
+        const res = await request(app).post(HOME_ROUTE).send({
+          name: "new supplier",
+        });
+
+        expect(res.statusCode).toBe(400);
+        expect(res.body).toHaveProperty("errorKey");
+        expect(res.body).toHaveProperty("errorDescription");
+        expect(res.body.errorKey).toBe("phone");
+        expect(res.body.errorDescription).toBe("phone is required");
+      });
+
+      it("Should return 400 if no address is sent to the route", async () => {
+        const res = await request(app).post(HOME_ROUTE).send({
+          name: "new supplier",
+          phone: "123456",
+        });
+
+        expect(res.statusCode).toBe(400);
+        expect(res.body).toHaveProperty("errorKey");
+        expect(res.body).toHaveProperty("errorDescription");
+        expect(res.body.errorKey).toBe("address");
+        expect(res.body.errorDescription).toBe("address is required");
+      });
+
+      it("Should return 400 if no password is sent to the route", async () => {
```

Comment:
Remove password validation methods, the Supplier does not have a password

Reply:
consider it done.